cmd/yggd/grpc.go

```diff
@@ -113,6 +116,41 @@ func (d *dispatcher) Send(ctx context.Context, r *pb.Data) (*pb.Receipt, error)
 	return &pb.Receipt{}, nil
 }
 
+func (d *dispatcher) sendEvents() {
+	for data := range d.eventsQ {
+		workers := []worker{}
+		d.RLock()
+		for _, wkr := range d.workers {
+			workers = append(workers, wkr)
+		}
+		d.RUnlock()
+		if len(workers) == 0 {
+			log.Warnf("cannot route message to workers: %v", data)
+			continue
+		}
+		f := func(wrk worker, data *pb.APIResponse) {
+			conn, err := grpc.Dial("unix:"+wrk.addr, grpc.WithInsecure())
+			if err != nil {
+				log.Errorf("cannot dial socket: %v", err)
+				return
+			}
+			defer conn.Close()
+
+			c := pb.NewWorkerClient(conn)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+
+			_, err = c.Events(ctx, data)
+			if err != nil {
+				log.Errorf("SendEvents: cannot send event to %v: %v", wrk.addr, err)
+			}
+		}
+		for _, wrk := range workers {
```

Comment:
My understanding of that code is that every event that is put on the eventQ (which is all the responses for all requests sent out to the control plane) is sent to every worker registered in yggdrasil - every request made by any worker causes N (number of workers)  events to be sent to every worker. 
Is that how that code works is there some implicit filtering/dispatching that I missed?

Not all workers are interested (or even should be) in all events - especially when the events are responses to requests. That also might be considered a security hole - exposing information directed to only one recipient to others. 

Only worker sending request that causes the event should receive the event.